agenta-web/src/components/pages/evaluations/NewEvaluation/SelectTestsetSection.tsx

```diff
@@ -0,0 +1,135 @@
+import {formatDate} from "@/lib/helpers/dateTimeHelper"
+import {testset} from "@/lib/Types"
+import {CloseCircleOutlined} from "@ant-design/icons"
+import {Collapse, Input, Space, Tag} from "antd"
+import Table, {ColumnsType} from "antd/es/table"
+import dayjs from "dayjs"
+import React, {useMemo, useState} from "react"
+
+type SelectTestsetSectionProps = {
+    testSets: testset[]
+    selectedTestsetId: string
+    setSelectedTestsetId: React.Dispatch<React.SetStateAction<string>>
+} & React.ComponentProps<typeof Collapse>
+
+const SelectTestsetSection = ({
+    testSets,
+    selectedTestsetId,
+    setSelectedTestsetId,
+    ...props
+}: SelectTestsetSectionProps) => {
+    const [searchTerm, setSearchTerm] = useState("")
+
+    const columns: ColumnsType<testset> = [
+        {
+            title: "Name",
+            dataIndex: "name",
+            key: "name",
+            onHeaderCell: () => ({
+                style: {minWidth: 220},
+            }),
+        },
+        {
+            title: "Date Modified",
+            dataIndex: "updated_at",
+            key: "updated_at",
+            onHeaderCell: () => ({
+                style: {minWidth: 220},
+            }),
+            render: (date: string) => {
+                return formatDate(date)
+            },
+        },
+        {
+            title: "Date created",
+            dataIndex: "created_at",
+            key: "created_at",
+            render: (date: string) => {
+                return formatDate(date)
+            },
+            onHeaderCell: () => ({
+                style: {minWidth: 220},
+            }),
+        },
+    ]
+
+    const filteredTestset = useMemo(() => {
+        let allTestsets = testSets.sort(
+            (a: testset, b: testset) =>
+                dayjs(b.updated_at).valueOf() - dayjs(a.updated_at).valueOf(),
+        )
+        if (searchTerm) {
+            allTestsets = testSets.filter((item: testset) =>
+                item.name.toLowerCase().includes(searchTerm.toLowerCase()),
+            )
+        }
+        return allTestsets
+    }, [searchTerm, testSets])
+
+    const handleRemoveTestset = () => {
+        setSelectedTestsetId("")
+    }
+
+    const selectedTestset = useMemo(
+        () => testSets.find((testset) => testset._id === selectedTestsetId) || null,
+        [selectedTestsetId, testSets],
+    )
+
+    return (
+        <Collapse
+            defaultActiveKey={["1"]}
+            {...props}
+            items={[
```

Comment:
let's improve this part. This is not good practice, because these'll be recreated each render. even if they don't change React will see the entire items prop as a new object and triggers unnecessary re-renders.